hbase-server/src/main/java/org/apache/hadoop/hbase/master/WorkerAssigner.java

```diff
@@ -60,27 +60,23 @@ public synchronized Optional<ServerName> acquire() {
       .findAny();
     worker.ifPresent(name -> currentWorkers.compute(name, (serverName,
       availableWorker) -> availableWorker == null ? maxTasks - 1 : availableWorker - 1));
+    event.suspend();
+    event.suspendIfNotReady(proc);
     return worker;
   }
 
   public synchronized void release(ServerName serverName) {
     currentWorkers.compute(serverName, (k, v) -> v == null ? null : v + 1);
-  }
-
-  public void suspend(Procedure<?> proc) {
-    event.suspend();
-    event.suspendIfNotReady(proc);
-  }
-
-  public void wake(MasterProcedureScheduler scheduler) {
     if (!event.isReady()) {
-      event.wake(scheduler);
+      event.wake(master.getMasterProcedureExecutor().getEnvironment().getProcedureScheduler());
     }
   }
 
   @Override
   public void serverAdded(ServerName worker) {
```

Comment:
Better to also add synchronized here?

Reply:
sorry, miss here. will fix it. Thanks for reminding.